integrations/utils.ts

```diff
@@ -51,68 +55,33 @@ export function test(
   name: string,
   config: TestConfig,
   testCallback: TestCallback,
-  { only = false } = {},
+  { only = false, debug = false }: TestFlags = {},
 ) {
   return (only ? defaultTest.only : defaultTest)(
     name,
     { timeout: TEST_TIMEOUT },
     async (options) => {
-      let root = await fs.mkdtemp(
-        // On Windows CI, tmpdir returns a path containing a weird RUNNER~1 folder
-        // that apparently causes the vite builds to not work.
-        path.join(
-          process.env.CI && platform() === 'win32' ? homedir() : tmpdir(),
-          'tailwind-integrations',
-        ),
-      )
-
-      async function write(filename: string, content: string): Promise<void> {
-        let full = path.join(root, filename)
-
-        if (filename.endsWith('package.json')) {
-          content = await overwriteVersionsInPackageJson(content)
-        }
-
-        // Ensure that files written on Windows use \r\n line ending
-        if (platform() === 'win32') {
-          content = content.replace(/\n/g, '\r\n')
-        }
-
-        let dir = path.dirname(full)
-        await fs.mkdir(dir, { recursive: true })
-        await fs.writeFile(full, content)
-      }
-
-      for (let [filename, content] of Object.entries(config.fs)) {
-        await write(filename, content)
-      }
-
-      try {
-        execSync('pnpm install', { cwd: root })
-      } catch (error: any) {
-        console.error(error.stdout.toString())
-        console.error(error.stderr.toString())
-        throw error
-      }
-
-      let disposables: (() => Promise<void>)[] = []
-
-      async function dispose() {
-        await Promise.all(disposables.map((dispose) => dispose()))
-        try {
-          await fs.rm(root, { recursive: true, maxRetries: 5, force: true })
-        } catch (err) {
-          if (!process.env.CI) {
-            throw err
-          }
-        }
+      let rootDir = debug
+        ? path.join(REPO_ROOT, '.debug')
+        : // On Windows CI, tmpdir returns a path containing a weird RUNNER~1
+          // folder  that apparently causes the vite builds to not work.
+          process.env.CI && platform() === 'win32'
+          ? homedir()
+          : tmpdir()
+      await fs.mkdir(rootDir, { recursive: true })
+
+      let root = await fs.mkdtemp(path.join(rootDir, 'tailwind-integrations'))
+
+      if (debug) {
+        console.log('Running test in debug mode. File system will be written to:')
+        console.log(root)
+        console.log()
       }
 
-      options.onTestFinished(dispose)
-
       let context = {
         root,
         async exec(command: string, childProcessOptions: ChildProcessOptions = {}) {
+          if (debug) console.log(`> ${command}`)
           return execSync(command, {
             cwd: root,
```

Comment:
While testing I ran into an issue that I was executing the commands in the wrong folder. If we pass in a custom `cwd`, I think we should show that as a step as well:
```suggestion
          let cwd = childProcessOptions.cwd ?? root
          if (debug && cwd !== root) {
            let relative = path.relative(root, cwd)
            if (relative[0] !== '.') relative = `./${relative}`
            console.log(`> cd ${relative}`)
          }
          if (debug) console.log(`> ${command}`)
          return execSync(command, {
            cwd,
```

This results in:
<img width="1242" alt="image" src="https://github.com/user-attachments/assets/a8198aee-2fdd-474c-a638-7cef5ef71f88">